src/components/WorkQueueIconText.vue

```diff
@@ -14,29 +14,48 @@
 </template>
 
 <script lang="ts" setup>
-  import { useSubscription, useSubscriptionWithDependencies } from '@prefecthq/vue-compositions'
+  import { useSubscriptionWithDependencies } from '@prefecthq/vue-compositions'
   import { computed } from 'vue'
   import { useWorkspaceApi, useWorkspaceRoutes } from '@/compositions'
 
   const props = defineProps<{
     workQueueName: string,
+    workPoolName?: string | null,
+    workPoolId?: string | null,
   }>()
 
   const api = useWorkspaceApi()
   const routes = useWorkspaceRoutes()
 
-  const workQueuesSubscription = useSubscription(api.workQueues.getWorkQueueByName, [props.workQueueName])
-  const workQueue = computed(() => workQueuesSubscription.response)
-
   const workPoolArgs = computed<Parameters<typeof api.workPools.getWorkPools> | null>(() => {
-    if (!workQueue.value?.workPoolId) {
+    if (!props.workPoolId) {
       return null
     }
 
-    return [{ 'work_pools': { id: { any_: [workQueue.value.workPoolId] } } }]
+    return [{ 'work_pools': { id: { any_: [props.workPoolId] } } }]
+
   })
 
   const workPoolsSubscription = useSubscriptionWithDependencies(api.workPools.getWorkPools, workPoolArgs)
   const workPools = computed(() => workPoolsSubscription.response ?? [])
-  const workPool = computed(() => workPools.value[0])
+  const workPoolName = computed(() => workPools.value[0].name)
```

Comment:
This could maybe be simplified so that you wouldn't need a separate `workPoolNameRef` variable
```suggestion
  const workPoolName = computed(() => pros.workPoolName ?? workPools.value[0].name)
```

Reply:
Nice, good suggestion! Thanks